cmd/dep/gopath_scanner.go

```diff
@@ -52,10 +52,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 		return err
 	}
 
-	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
-	}
+	g.origM = dep.NewManifest()
+	g.origM.Constraints = g.pd.constraints
```

Comment:
The cases which overwrite the `Constraints` map in this way now have an extra initialization in the constructor.  Multiple constructors or multiple optional paramaters seem like messy options.  What about something like an `Init()` (or `SetDefaults` etc.) method instead, which only inits nil maps?